app/activity/[id]/components/activity-card.tsx

```diff
@@ -0,0 +1,82 @@
+'use client'
+
+import { Heading, Box, Flex, Spacer, Text, IconButton } from '@chakra-ui/react'
+import { FiClock, FiMapPin, FiLink2, FiEdit3, FiTrash2 } from 'react-icons/fi'
+import { Link } from '@/components/link'
+import { customColors } from '@/theme/color'
+import { ActivityCollectionQuery } from '@generated/api'
+
+type ActivityInfoProps = {
+  activityData: ActivityCollectionQuery | undefined
+}
+
+export const ActivityCard: React.FC<ActivityInfoProps> = ({ activityData }) => {
+  if (!activityData || !activityData.activityCollection) {
+    return null
+  }
```

Comment:
I'd say we don't have to specify this condition due to the reason that I mentioned in the other part.